os/kernel/init/os_smpstart.c

```diff
@@ -79,8 +79,10 @@ void os_idle_trampoline(void)
 
 	for (; ; ) {
 		/* Perform any processor-specific idle state operations */
-
-		up_idle();
+		/* Let primary core control the PM idle loop, other 
+		   cores may stay at WFE and wait for signal from primary core
+		*/
+		SP_WFE();
```

Comment:
It looks good to me.

If second core entering the wfe, power can be reduced than two core is working?

Reply:
Yes, the power consumption should be as below:
(primary core running + secondary core wife) < (primary core running + secondary core running)
But, the core task assignment depends on kernel SMP mechanism.